### PR TITLE
Remove the dockerlab VM

### LINKED ISSUE #58: Lab 1 - Remove dockerlab VM
Let them use Docker on their computer, not in a VM.

### PR DESCRIPTION
**Pull request by thomasaelbrecht**
Closes #58

### CODE REVIEW

File: assignments/01-docker.md [hunk 1]
```diff
@@ -29,29 +29,25 @@ In this lab assignment, you will refresh your knowledge of Docker. You will crea
...
-```console
-cd dockerlab
-vagrant up dockerlab
-```
-
-> If you're have issues with Vagrant, you may use Docker on your local machine. In that case, you don't need the VM we're using in this lab assignment. You can simply destroy it using `vagrant destroy -f`.
-> Make sure to start Portainer yourself then! There is `docker-compose.yml` file in the `dockerlab/provisioning/dockerlab` folder that you can use to start Portainer.
+- Windows: use [Docker Desktop](https://docs.docker.com/desktop/) in combination with the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install)
+- macOS: use [Docker Desktop](https://docs.docker.com/desktop/)
+- Linux: use [Docker Engine](https://docs.docker.com/engine/install/) and **not** Docker Desktop
```
Comment (MartenBE) [inline assignments/01-docker.md:36]: Perhaps we can also advise them to use the package manager and just check if the version is up to date (or recent enough).

File: assignments/01-docker.md [hunk 2]
```diff
@@ -110,13 +108,13 @@ Make sure that the application can connect to the database by setting the `MYSQL
 >
 > :bulb: There is also an HTTP header `X-Database-Used` in every response that indicates which database is used.
 
-You could bind the MySQL port (3306) to a port on your VM, but it's a bad idea to expose your database to the outside world. You should only allow traffic on the port of your web service, the database doesn't need public access. You should block all other incoming traffic by using a firewall, but this is outside the scope of this course. However if you bind the database port to a port on your VM, you can use SSH port forwarding to access the database from your own pc (over an SSH connection).
+In a real-world scenario, you could bind the MySQL port (3306) to a port on your VM, but it's a bad idea to expose your database to the outside world. You should only allow traffic on the port of your web service, the database doesn't need public access. You should block all other incoming traffic by using a firewall, but this is outside the scope of this course. However if you bind the database port to a port on your VM, you can use SSH port forwarding to access the database from your own pc (over an SSH connection).
```
Comment (MartenBE) [inline assignments/01-docker.md:111]: Remove the mention of a VM. Also, in a real-world scenario would actually be the scenario where this should not happen. I think we better rephrase here.

**Review by MartenBE — CHANGES_REQUESTED** [review]

File: assignments/01-docker.md (hunk 2 above)
Comment (thomasaelbrecht) [reply assignments/01-docker.md:111]: Something like this?

In a real-world scenario, it's a bad idea to expose your database to the outside world. You should only allow traffic from specific domains or IP addresses. You should block all other incoming traffic by using a firewall, but this is outside the scope of this course.

File: assignments/01-docker.md (hunk 1 above)
Comment (thomasaelbrecht) [reply assignments/01-docker.md:36]: Should we mention the right commands or do they need to figure them out?

File: assignments/01-docker.md (hunk 2 above)
Comment (thomasaelbrecht) [reply assignments/01-docker.md:111]: This is better:

In a real-world scenario, it's a bad idea to expose your database to the outside world. First think if public access is really necessary. If so, you should only allow traffic from specific domains or IP addresses. You should block all other incoming traffic by using a firewall, but this is outside the scope of this course.

File: assignments/01-docker.md (hunk 1 above)
Comment (MartenBE) [reply assignments/01-docker.md:36]: Let them figure it out, but perhaps mention `apt`, `dnf`, `pacman`, ... and that they have to look it up according to their distro of choice. They should be able to handle that now, they've had Linux for Datascientists after all.

Comment (thomasaelbrecht) [reply assignments/01-docker.md:36]: Isn't this already the case? Or should we remove the links for all OS'es?

Comment (MartenBE) [reply assignments/01-docker.md:36]: Yes, but I would just also add the following:

> ... through a package manager (e.g. `apt`, `dnf`, `pacman`, ... depending on your distro). ...

Comment (thomasaelbrecht) [reply assignments/01-docker.md:36]: https://github.com/HOGENT-MLOps/mlops-labs/blob/da1e8ed3fcb4bc098d9dfc2297a34dbc0d1e193f/assignments/01-docker.md?plain=1#L32

**Review by MartenBE — APPROVED** [review]

**Pull request merged**